packages/cloud_firestore/lib/src/query_snapshot.dart

```diff
@@ -34,5 +38,7 @@ class QuerySnapshot {
   /// is the first snapshot, all documents will be in the list as Added changes.
   final List<DocumentChange> documentChanges;
 
+  final SnapshotMetadata metadata;
```

Comment:
We should be able to add unit testing and a basic integration test for this, right? Even just asserting that `hasPendingWrites` and `isFromCache` aren't null would be better than nothing.

Reply:
Done, I checked that the metadata property of the QuerySnapshot was not null.